src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs

```diff
@@ -47,12 +47,12 @@ protected override bool ReceiveRecover(object message)
             if (message is EntityStarted)
             {
                 var started = (EntityStarted)message;
-                State = new ShardState(State.Entries.Add(started.EntityId));
+                State = new ShardState(State.Entries?.Add(started.EntityId));
```

Comment:
I think it will be easier to make the change in the `ShardState` constructor instead of null checking everytime:

```csharp
public ShardState(IImmutableSet<EntityId> entries) : this()
{
    Entries = entries ?? ImmutableHashSet<string>.Empty;
}
```